### PR TITLE
Update torch-xpu-ops commit pin

### PR DESCRIPTION
**Pull request by toyxu**
Update the torch-xpu-ops commit to [5e29831 ](https://github.com/intel/torch-xpu-ops/commit/5e29831). Includes: 
- OneAPI-2025 build issue fix
- Enhancement of the XPU operator coverage

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by EikanWang — COMMENTED** [review]
@xytintel , pls. refine the pr description a little bit. Although touch-put-ops is out of tree. A good description will help us track the update.

**Comment by EikanWang** [comment]
@xytintel , pls. update the test cases accordingly.

**Comment by xytintel** [comment]
Depend on https://github.com/pytorch/pytorch/pull/140109

**Review by guangyey — APPROVED** [review]

File: test/inductor/test_torchinductor_opinfo.py [hunk 1]
```diff
@@ -352,8 +352,6 @@ def format_op(op):
     # rrelu not supported on XPU now
     "nn.functional.rrelu": {f16, f32, f64},
     # not implemented for 'Half'
-    "nn.functional.multilabel_margin_loss": {f16},
-    "nn.functional.multi_margin_loss": {f16},
```
Comment (guangyey) [inline test/inductor/test_torchinductor_opinfo.py:356]: We remove them from expected failure as we support f16 data type on these ops in torch-xpu-ops.

**Review by guangyey — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**